src/main/java/minijava/ir/optimize/ConstantFolder.java

```diff
@@ -185,4 +185,33 @@ public void visit(Shrs node) {
   public void visit(Sub node) {
     visitBinaryOperation((lhs, rhs) -> lhs.sub(rhs), node, node.getLeft(), node.getRight());
   }
+
+  @Override
+  public void visit(Phi node) {
+    int n = node.getPredCount();
```

Comment:
![clarifying svg](http://imgh.us/control-flow-sensitive-consts.svg)

If you look at the inputs to phi node 80 in block 69, there is this reference to Cmp 73.
We know from the incoming control flow edge 0 that that must evaluate constantly to false. This is what this does. I think I'll move it into a separate pass.